### DISCUSSION TITLE
Use OR instead of AND condition

### DISCUSSION THREAD

**Discussion by JeromeK13** [carltongibson/django-filter#1426] [Q&A]
Im currently using the following filter with DRF:
But when i filter it will be always using an AND condition how can i change this schema to support OR conditons?
```python
class SystemFilter(djfilters.FilterSet):
    id_v = djfilters.filters.Filter(field_name='id', lookup_expr='icontains')
    system= djfilters.filters.Filter(field_name='system', lookup_expr='icontains')
    deputy = djfilters.filters.Filter(field_name='deputy, lookup_expr='icontains')
    deputy_2 = djfilters.filters.Filter(field_name='deputy_2', lookup_expr='icontains')

   class Meta:
        model = System
        fields = ['id_v', 'system', 'deputy', 'deputy_2']


class SystemAPIViewSet(viewsets.ModelViewSet):
    queryset = System.objects.all()
    permission_classes = [IsAdminOrReadOnly]
    serializer_class = SystemSerializer
    pagination_class = DynamicPageSize
    filter_backends = [django_filters.rest_framework.DjangoFilterBackend, filters.OrderingFilter]
    filterset_class = SystemFilter

```

**Comment by sunbinghonter** [comment]
class SystemFilter(djfilters.FilterSet):
    id_v = djfilters.filters.Filter(field_name='id', lookup_expr='icontains')
    system= djfilters.filters.Filter(field_name='system', lookup_expr='icontains')
    deputy = djfilters.filters.Filter(field_name='deputy, lookup_expr='icontains')
    deputy_2 = djfilters.filters.Filter(field_name='deputy_2', lookup_expr='icontains')
    deputys = djfilters.charFilter(method='my_deputys_filter')

   class Meta:
        model = System
        fields = ['id_v', 'system', 'deputy', 'deputy_2']

  def my_deputys_filter(self,queryset,name,value):
        return queryset.filter(
                          Q(deputy=value)|
                          Q(deputy2=value)
                   )
  try this. it is working in my project.

**Comment by reddytocode** [comment]
I've created this simple way for anyone still trying to do this

For DRF, add this to your FilterSet class, it will transform your query to use OR instead of AND
```
    def filter_queryset(self, queryset):
        filter_queries = []
        for name, value in self.form.cleaned_data.items():
            if value is not None:
                qs = self.filters[name].filter(queryset, value)
                filter_queries.append(qs)

        res = queryset
        for index, q in enumerate(filter_queries):
            if index == 0:
                res = q
            else:
                res = res | q
        return res
```

**Comment by giuseppenovielli** [comment]
https://github.com/philipn/django-rest-framework-filters?tab=readme-ov-file#complex-querystring-encoding

But isn't good maintained!
